### PR TITLE
[CI] Extract interpreter workflow and split `std_spec` execution

### PR DESCRIPTION
**Pull request by straight-shoota**
`test_interpreter` is failing almost constantly in CI due to running out of memory.

Execution of `interpreter-std_spec` uses almost 10G memory, but the GHA runners only have around 8G available.
```
        Command being timed: "bin/crystal i spec/interpreter_std_spec.cr -- --junit_output .junit/interpreter-std_spec.xml"
        User time (seconds): 369.31
        System time (seconds): 4.90
        Percent of CPU this job got: 277%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 2:15.03
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 9926492
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: 686100
        Voluntary context switches: 21253
        Involuntary context switches: 6934
        Swaps: 0
        File system inputs: 0
        File system outputs: 6128
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 0
```
Splitting the entire spec suite into separate pieces reduces the total amount of memory. We already have the mechanism with `SPEC_SPLIT` available. It was previously used for CI on x86.

We should probably try to reduce the interpreters memory usage, but this is a quick workaround to get CI green again.

### CODE REVIEW

File: .github/workflows/interpreter.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Interpreter Test
```
Comment (Sija) [inline .github/workflows/interpreter.yml:1]: Naming suggestion to keep consistency with the rest of the workflows.

```suggestion
name: Interpreter CI
```

File: .github/workflows/interpreter.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  SPEC_SPLIT_DOTS: 160
+
+jobs:
+  test-interpreter_spec:
```
Comment (Sija) [inline .github/workflows/interpreter.yml:13]: Naming consistency-wise:

```suggestion
  test-interpreter-spec:
```

File: .github/workflows/interpreter.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    runs-on: ubuntu-22.04
+    container:
+      image: crystallang/crystal:1.7.3-build
+    name: "Test Interpreter"
```
Comment (Sija) [inline .github/workflows/interpreter.yml:17]: I'd avoid redundant naming (workflow name points at the Interpreter already)

```suggestion
    name: Test
```

File: .github/workflows/interpreter.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Test interpreter_spec
+        run: make deps && bin/crystal build -o interpreter_spec spec/compiler/interpreter_spec.cr && ./interpreter_spec --junit_output .junit/interpreter_spec.xml
+
+  build-interpreter:
+    runs-on: ubuntu-22.04
+    container:
+      image: crystallang/crystal:1.7.3-build
+    name: Build interpreter
```
Comment (Sija) [inline .github/workflows/interpreter.yml:28]: ditto

```suggestion
    name: Build
```

**Review by Sija — COMMENTED** [review]
Several naming suggestions

File: .github/workflows/interpreter.yml (hunk 2 above)
Comment (straight-shoota) [reply .github/workflows/interpreter.yml:13]: The spec suite that's being run is called `interpreter_spec`.

**Review by beta-ziliani — APPROVED** [review]
LGTM

**Pull request merged**